routes/adjustmentMethods/EToAdjustmentMethod.ts

```diff
@@ -0,0 +1,224 @@
+import * as SunCalc from "suncalc";
+import * as moment from "moment";
+import { AdjustmentMethod, AdjustmentMethodResponse, AdjustmentOptions } from "./AdjustmentMethod";
+import { GeoCoordinates, WateringData, WeatherProviderId } from "../../types";
+import { WeatherProvider } from "../weatherProviders/WeatherProvider";
+
+
+/**
+ * Calculates how much watering should be scaled based on weather and adjustment options by comparing the recent
+ * potential ETo to the baseline potential ETo that the watering program was designed for.
+ */
+async function calculateEToWateringScale(
+	adjustmentOptions: EToScalingAdjustmentOptions,
+	wateringData: WateringData | undefined,
+	coordinates: GeoCoordinates,
+	weatherProvider: WeatherProvider
+): Promise< AdjustmentMethodResponse > {
+
+	// Temporarily disabled since OWM forecast data is checking if rain is forecasted for 3 hours in the future.
+	/*
+	if ( wateringData && wateringData.raining ) {
+		return {
+			scale: 0,
+			rawData: { raining: 1 }
+		}
+	}
+	 */
+
+	// This will throw an error message if ETo data cannot be retrieved.
+	const etoData: EToData = await weatherProvider.getEToData( coordinates );
+
+	let baseETo: number;
+	// Default elevation is based on data from https://www.pnas.org/content/95/24/14009.
+	let elevation = 600;
+
+	if ( adjustmentOptions && "baseETo" in adjustmentOptions ) {
+		baseETo = adjustmentOptions.baseETo
+	} else {
+		throw "A baseline potential ETo must be provided.";
+	}
+
+	if ( adjustmentOptions && "elevation" in adjustmentOptions ) {
+		elevation = adjustmentOptions.elevation;
+	}
+
+	const eto: number = calculateETo( etoData, elevation, coordinates );
+
+	const scale =  Math.floor( Math.min( Math.max( 0, ( eto - etoData.precip ) / baseETo * 100 ), 200 ) );
+	return {
+		scale: scale,
+		rawData: {
+			eto: Math.round( eto * 1000) / 1000,
+			radiation: Math.round( etoData.solarRadiation * 100) / 100
+		}
+	}
+}
+
+/* The implementation of this algorithm was guided by a step-by-step breakdown
```

Comment:
Do you know if the temperature and humidity provided by OWM and Dark Sky at hourly intervals are averages for that interval or values at the start of that interval? The documentation is a bit ambiguous, and we might lose some accuracy if they're just instantaneous values

Reply:
DarkSky is explicitly average (or sum for precip etc). OWM is, by observation, average.

Reply:
I'll implement the hourly equation and see how the results compare to ones from the daily equation

Reply:
I did a little bit of research into the differences between hourly and daily ETo calculations, and I think that we may want to leave the equation as-is for now. Although the hourly step may yield a more accurate result, I think it is more important that we try to measure the ETo in the same way that the baseline ETo is measured. If we calculate the ETo with an hourly equation and get a result that is consistently lower than the result of the daily equation but use the daily equation to calculate the baseline ETo, it will result in watering scales that are consistently lower than they should be. Most of the baseline ETo data sources I have found ([MOD16](https://www.ntsg.umt.edu/project/modis/mod16.php), [Australian BoM](http://www.bom.gov.au/watl/eto/), [CIMIS](https://cimis.water.ca.gov/)) use daily calculations rather than hourly calculations, so I think we should do the same.